src/richie/apps/courses/models/blog.py

```diff
@@ -30,6 +31,52 @@ def __str__(self):
             name=self.extended_object.get_title(), model=self._meta.verbose_name.title()
         )
 
+    def _get_category_pages(self, language=None):
+        """
+        Return the category pages linked to the blogpost via a category plugin in the "categories"
+        placeholder on the blogpost detail page, ranked by their `path` to respect the
+        order in the categories tree.
+        """
+        language = language or translation.get_language()
+
+        selector = "category_plugins__cmsplugin_ptr"
+        # pylint: disable=no-member
+        filter_dict = {
+            f"{selector:s}__language": language,
+            f"{selector:s}__placeholder__page": self.extended_object,
+        }
+        # For a public blogpost, we must filter out categories that are not published in
+        # any language
+        if self.extended_object.publisher_is_draft is False:
+            filter_dict["title_set__published"] = True
+
+        return Page.objects.filter(**filter_dict).order_by("node__path").distinct()
+
+    def get_related_blogposts(self, language=None):
+        """
+        Return the blogposts related to the current blogpost ie that share common categories.
+        the blogposts returned are sorted by their `path` to respect the order in the page
+        tree.
+        """
+        is_draft = self.extended_object.publisher_is_draft
+        language = language or translation.get_language()
+
+        bfs = "extended_object__placeholders__cmsplugin__courses_categorypluginmodel__page__in"
+        selector = {bfs: self._get_category_pages()}
+
+        # pylint: disable=no-member
+        return (
+            BlogPost.objects.filter(
+                extended_object__publisher_is_draft=is_draft,
+                extended_object__placeholders__cmsplugin__language=language,
+                **selector,
+            )
+            .exclude(id=self.id)
+            .select_related("extended_object")
+            .distinct()
+            .order_by("extended_object__node__path")
```

Comment:
It's just a suggestion here, but should we generally order blog posts by date instead of node path?

Reply:
On the "blogpost" main page they are sorted by publication date... but here I thought sorting by path would allow changing the order to convey importance.
Anyway, if you don't voluntarily change the order, the blogposts will be sorted by date because they are created when you want to publish them...

Reply:
But you would have to reorder them in the tree when that does not affect their order on the blogpost page. That does not seem very intuitive to me.

> Anyway, if you don't voluntarily change the order, the blogposts will be sorted by date because they are created when you want to publish them...

Maybe I'm just nitpicking then. This is ok 😅

Reply:
We could decide to order the blogpost page by importance as well... This is a very small change in the template. We could make it a setting if it proves necessary. We will see what the users say.